### PR TITLE
Balder/quick restart of slobrok

### PR DESCRIPTION
**Pull request by baldersheim**
@arnej27959 PR
@hmusum FYI

### CODE REVIEW

**Comment by baldersheim** [comment]
@havardpe This might be of interest to you too.

File: slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+namespace slobrok {
+
+ReconfigurableStateServer::ReconfigurableStateServer(const config::ConfigUri & configUri,
+                                                     vespalib::HealthProducer & health,
+                                                     vespalib::MetricsProducer & metrics,
+                                                     vespalib::ComponentConfigProducer & components)
+    : _health(health),
+      _metrics(metrics),
+      _components(components),
+      _configFetcher(std::make_unique<config::ConfigFetcher>(configUri.getContext())),
```
Comment (havardpe) [inline slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp]: you probably want to unsubscribe before deleting the server

File: slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    _server.reset();
+    for (size_t retryTime(1); !_server && (retryTime < 10); retryTime++) {
+        try {
+            _server = std::make_unique<vespalib::StateServer>(config->httpport, _health, _metrics, _components);
+        } catch (vespalib::PortListenException & e) {
+            LOG(warning, "Failed listening to network port(%d) with protocol(%s): '%s', will retry for 60s",
+                e.get_port(), e.get_protocol().c_str(), e.what());
+            std::this_thread::sleep_for(retryTime * 1s);
+        }
+    }
```
Comment (havardpe) [inline slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp]: what to do when everything fails?

Comment (havardpe) [reply slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp]: which thread will the retrying block?

Comment (baldersheim) [reply slobrok/src/vespa/slobrok/server/reconfigurable_stateserver.cpp]: 1 - std::quick_exit
2 - config fetcher thread.

**Comment by baldersheim** [comment]
Factory says: Not good enough !

**Pull request merged**